### ISSUE TITLE
Ruff fails to detect duplicate type parameter names

### ISSUE THREAD

**Issue by JelleZijlstra** [astral-sh/ruff#11119]
This is a SyntaxError:

```
>>> type X[T, T] = int
  File "<stdin>-0", line 1
SyntaxError: duplicate type parameter 'T'
```

But Ruff is fine with it. Similar to #11118.

**Comment by dhruvmanila** [comment]
We do raise a similar error for function parameters (https://play.ruff.rs/065d0212-7f56-4fc1-b5fc-44d781c288fa) although I would prefer to move this into the linter once #1774 is done. The reason it exists today is for backwards compatibility with the old parser where it did raise this error during parsing.
